### PR TITLE
Fix some `useUserThreads` issues

### PR DESCRIPTION
**Pull request by nvie**
This PR:

- Contains some pure refactorings (moving code around, renaming functions, etc)
- DRYs up the compareThread logic we had defined in multiple places
- Fix bug in metadata filtering

### CODE REVIEW

File: packages/liveblocks-react/src/liveblocks.tsx [hunk 1]
```diff
@@ -109,17 +112,74 @@ function selectorFor_useInboxNotifications(
...
+  );
+
+  // Second filter pass: select only threads matching query filter
+  const query = options.query;
+  if (query) {
+    threads = threads.filter(makeThreadsFilter(query));
+  }
+
+  // Sort threads by updated date (newest first) and then created date
+  return threads.sort(byMostRecentlyUpdated);
```
Comment (FlowFlorent) [inline packages/liveblocks-react/src/liveblocks.tsx:174]: that reads nicely 🙌

File: packages/liveblocks-react/src/umbrella-store.ts [hunk 2]
```diff
@@ -936,12 +905,13 @@ export function applyThreadUpdates<M extends BaseMetadata>(
   updates.newThreads.forEach((thread) => {
     const existingThread = updatedThreads[thread.id];
 
-    // If the thread already exists, we need to compare the two threads to determine which one is newer.
+    // If a thread already exists and it's never, don't touch it
```
Comment (FlowFlorent) [inline packages/liveblocks-react/src/umbrella-store.ts:908]:
```suggestion
    // If a thread already exists and it's never, don't touch it
```
not sure what this sentence means?

**Review by FlowFlorent — APPROVED** [review]
Looks good! Just one small comment about a [comment](https://github.com/liveblocks/liveblocks/pull/1928/files#r1757395765)

File: packages/liveblocks-react/src/umbrella-store.ts (hunk 2 above)
Comment (nvie) [reply packages/liveblocks-react/src/umbrella-store.ts:908]: LOL, me neither 😂

**Review by FlowFlorent — APPROVED** [review]
🚀

**Pull request merged**